ios/Exponent/Versioned/Modules/Api/EXContacts.m

```diff
@@ -54,13 +67,54 @@ @implementation EXContacts
 
     NSMutableDictionary *contact = [NSMutableDictionary dictionary];
     contact[@"id"] = person.identifier;
+    contact[@"contactType"] = person.contactType == CNContactTypePerson ? @"person" : @"organization";
+    contact[@"name"] = [self _assembleDisplayNameForContact:person];
     contact[@"firstName"] = person.givenName;
     contact[@"lastName"] = person.familyName;
+    contact[@"previousLastName"] = person.previousFamilyName;
     contact[@"middleName"] = person.middleName;
-    contact[@"name"] = [self _assembleDisplayNameForContact:person];
+    contact[@"nickname"] = person.nickname;
     contact[@"company"] = person.organizationName;
     contact[@"jobTitle"] = person.jobTitle;
+    contact[@"department"] = person.departmentName;
+    contact[@"imageAvailable"] = [NSNumber numberWithBool:person.imageDataAvailable];
 
+    if ([keysToFetch containsObject:CNContactNamePrefixKey]) {
+      contact[@"namePrefix"] = person.namePrefix;
+    }
+    if ([keysToFetch containsObject:CNContactNameSuffixKey]) {
+      contact[@"nameSuffix"] = person.nameSuffix;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticGivenNameKey]) {
+      contact[@"phoneticFirstName"] = person.phoneticGivenName;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticMiddleNameKey]) {
+      contact[@"phoneticMiddleName"] = person.phoneticMiddleName;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticFamilyNameKey]) {
+      contact[@"phoneticLastName"] = person.phoneticFamilyName;
+    }
+    if ([keysToFetch containsObject:CNContactNoteKey]) {
+      contact[@"note"] = person.note;
+    }
+    if ([keysToFetch containsObject:CNContactImageDataKey]) {
+      contact[@"image"] = @{ @"uri": [NSString stringWithFormat:@"%@%@",
```

Comment:
@brentvatne together with @aalices we decided we're going to add a method to the JS contacts API that would allow you to fetch contacts data for a single contact. This way you could do a superset of things you can with `getImageUriForContactAsync` (imagine you want to implement contacts app where you list contacts by name on the main view and then when you navigate into details you see all the data, currently contacts API only have one method to fetch all contacts).

On top of that we are going to log a warning (yellow box) whenever you try to fetch image data for a list of contacts and suggest you should only do that for a single contact.

@aalices will be updating this PR soon